ctwrap/modules/adiabatic_flame.py

```diff
@@ -114,19 +114,27 @@ def run(name, chemistry=None, upstream=None, domain=None, loglevel=0):
     return out
 
 
-def save(filename, data, task=None):
+def save(filename, data, task=None,  groups=None, **kwargs):
     """
     This function saves the output from the run method
 
     Arguments:
         filename (str): naming of file
         data (Dict): data to be saved
         task (str): name of task if running variations
+        groups (list): group names if file exists
```

Comment:
shouldn't groups be automatically 'inherited' from data keys?

Reply:
**PS:** `data.keys()` should be based on `task` (can be augmented like in `adiabatic_flame.py`)

Reply:
The ‘mode’ argument in ‘write_hdf’ applies to the file and not group. The ‘groups’ are the actual keys in the file, if the file exists.

Reply:
It's too confusing - the dictionary keys should be the same as the group keys in the file.

Reply:
What I am trying to do is this, if the `filename` exists, the file has been opened in `simulation` and the existing groups is passed here, so we can compare with `data` keys to know if it exists. I just don't want to include another dependency here - i.e, `h5py`.

Reply:
Ah. But you can check from `Simulation`, right?

Reply:
No,  For sure checking from the simulation modules is the better approach, but you were against it earlier. You want the save method to be short and sweet.

Reply:
I am still against it. What I mean is that you can check what is in a HDF file within `Simulation.save`. I probably have to look at details one more time, but I think it would be good to avoid too many task/group/key labels.

Reply:
There are many approaches to it, but the `mode` should be change on a group basis.

1. What I have currently is to check the existing group and pass it as an argument to the `save` method in the simulation modules.

2. Another approach is to do the checking in `simulation` and call the `save` method in the simulation modules in a loop.

Reply:
See MS teams:
> I think there should be one HDF file per batch job, that contains only information from that batch job. Which means:
> * if file doesn't exist, everything is fine
> * if file already exists, default is to raise an error
> * if file exists and `force=True` is specified, the file is deleted and recreated